onnx_caffe2/backend.py

```diff
@@ -297,7 +300,71 @@ def _create_gemm(cls, n, opset_version):
         return ops
 
     @classmethod
-    def _create_pad(cls, n, opset_version):
+    def _create_lstm(cls, onnx_model, n, opset_version):
+        assert onnx_model is not None, "cannot convert LSTMs through this codepath"
+
+        attrs = dict(n.attrs)
+        hidden_size = attrs.pop('hidden_size')
+        assert not attrs, "unsupported LSTM attributes: " + attrs.keys()
+
+        input_blob, W, R, B, sequence_lens, initial_h, initial_c = n.inputs
+
+        # ad-hoc, informally-specified, bug-ridden, slow
+        # implementation of shape inference
+        for x in onnx_model.graph.input:
+            if x.name == input_blob:
+                input_size = x.type.tensor_type.shape.dim[2].dim_value
+        for x in map(OnnxNode, onnx_model.graph.node):
+            if x.op_type == n.op_type and x.outputs[0] == input_blob:
+                input_size = x.attrs['hidden_size']
+
+        model_helper = ModelHelper()
+        name = dummy_name()
+
+        # input and recurrence biases are squashed together in onnx but not in caffe2
+        model_helper.param_init_net.Slice(B, name + "_bias_i2h",   starts=[0],             ends=[4*hidden_size])
+        model_helper.param_init_net.Slice(B, name + "_bias_gates", starts=[4*hidden_size], ends=[8*hidden_size])
+
+        # caffe2 has a different order from onnx. We need to rearrange
+        #   i o f c -> i f o c
+        def reform(name_from, name_to, extra_starts, extra_ends):
+            for start, suffix in enumerate(("_i", "_o", "_f", "_c")):
+                model_helper.param_init_net.Slice(
+                    name_from,
+                    name_from + suffix,
+                    starts = [   start  * hidden_size] + extra_starts,
+                    ends =   [(1+start) * hidden_size] + extra_ends)
+            model_helper.param_init_net.Concat(
+                [name_from + suffix for suffix in ("_i", "_f", "_o", "_c")],
+                [name_to, dummy_name()], axis=0)
+        reform(W, name + "/i2h_w", [0], [input_size])
```

Comment:
it's not obvious at all that those names match the ones constructed by LSTMCell and might accidentally break. It seems that if you pass 'return_params=True' to rnn_cell.LSTM() it'd return you `input` and `recurrent` params that are more of a standard API. Would it be sufficient for your needs?

Reply:
++ I'll try return_params